Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,875 @@
+- (void) _setSourceController: (id)controller;
+@end
+
+@interface NSStoryboardSegue (__StoryboardPrivate__)
+// Private to this class...
+- (void) _setKind: (NSString *)k;
+- (void) _setRelationship: (NSString *)r;
+- (NSString *) _kind;
+- (NSString *) _relationship;
+@end
+
+// this needs to be set on segues
+@implementation NSStoryboardSegue (__StoryboardPrivate__)
+- (void) _setKind: (NSString *)k
+{
+  ASSIGN(_kind, k);
+}
+
+- (void) _setRelationship: (NSString *)r
+{
+  ASSIGN(_relationship, r);
+}
+
+- (NSString *) _kind
+{
+  return _kind;
+}
+
+- (NSString *) _relationship
+{
+  return _relationship;
+}
+@end
+
+@implementation NSStoryboardSeguePerformAction
+- (id) target
+{
+  return _target;
+}
+
+- (void) setTarget: (id)target
+{
+  ASSIGN(_target, target);
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (NSString *) selector
+{
+  return NSStringFromSelector(_action);
+}
+
+- (void) setSelector: (NSString *)s
+{
+  _action = NSSelectorFromString(s);
+}
+
+- (id) sender
+{
+  return _sender;
+}
+
+- (void) setSender: (id)sender
+{
+  ASSIGN(_sender, sender);
+}
+
+- (NSString *) identifier
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (NSStoryboard *) storyboard
+{
+  return _storyboard;
+}
+
+- (void) setStoryboard: (NSStoryboard *)storyboard
+{
+  ASSIGN(_storyboard, storyboard);
+}
+
+- (NSStoryboardSegue *) storyboardSegue
+{
+  return _storyboardSegue;
+}
+
+- (void) setStoryboardSegue: (NSStoryboardSegue *)ss
+{
+  ASSIGN(_storyboardSegue, ss);
+}
+
+- (void) dealloc
+{
+  RELEASE(_storyboard);
+  RELEASE(_kind);
+  RELEASE(_identifier);
+  RELEASE(_sender);
+  RELEASE(_storyboardSegue);
+  [super dealloc];
+}
+
+- (IBAction) doAction: (id)sender
+{
+  BOOL should = YES;
+
+  // If the instance we are testing is a controller, then the value of should is set by this method....
+  // if it is not, as it is possible to initiate a segue from an NSMenuItem, then we don't, but should
+  // remains set to YES so that the logic to replace the destination controller is still called.
+  if ([_sender respondsToSelector: @selector(shouldPerformSegueWithIdentifier:sender:)])
+    {
+      should = [_sender shouldPerformSegueWithIdentifier: _identifier
+                                                  sender: _sender];
+    }
+
+  if (should)
+    {
+      id destCon = nil;
+      if ([[_storyboardSegue destinationController] isKindOfClass: [NSViewController class]] ||
+          [[_storyboardSegue destinationController] isKindOfClass: [NSWindowController class]])
+        {
+          destCon = [_storyboardSegue destinationController];
+        }
+      else
+        {
+          NSString *destId = [_storyboardSegue destinationController];
```

Comment:
Only in this case is the destination controller ever changed. It would be clearer if you checked the destination controller for being of class NSString and handled only that case:

if ([[_storyboardSegue destinationController] isKindOfClass: [NSViewController class]])
{
           NSString *destId = [_storyboardSegue destinationController];
destCon = [_storyboard instantiateControllerWithIdentifier: destId]; 
 // replace with actual controller...
[_storyboardSegue _setDestinationController: destCon]; 
         }

Reply:
Ah, quite true.

Reply:
Optimized this.